hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneManagerSnapshotAcl.java

```diff
@@ -382,6 +383,69 @@ public void testGetAclWithNotAllowedUser(BucketLayout bucketLayout)
     assertDoesNotThrow(() -> ozoneManager.getAcl(keyObj));
   }
 
+  @ParameterizedTest
+  @EnumSource(BucketLayout.class)
+  public void testLookupKeyWithAllowedUserForPrefixAcl(BucketLayout bucketLayout) throws Exception {
+    UserGroupInformation.setLoginUser(UGI1);
+
+    createVolume();
+
+    final OzoneVolume volume = objectStore.getVolume(volumeName);
+    createBucket(bucketLayout, volume);
+
+    final OzoneBucket bucket = volume.getBucket(bucketName);
+
+    setDefaultPrefixAcls();
+
+    createKey(bucket);
+
+    setDefaultVolumeAcls();
+    setDefaultBucketAcls();
+
+    createSnapshot();
+
+    final OmKeyArgs snapshotKeyArgs = getOmKeyArgs(true);
+    assertDoesNotThrow(() -> ozoneManager.lookupKey(snapshotKeyArgs));
+  }
+
+  @ParameterizedTest
+  @EnumSource(BucketLayout.class)
+  public void testLookupKeyWithNotAllowedUserForPrefixAcl(BucketLayout bucketLayout) throws Exception {
```

Comment:
nit
```suggestion
  public void testLookupKeyWithDisallowedUserForPrefixAcl(BucketLayout bucketLayout) throws Exception {
```

Reply:
Initially I used `Disallowed` but then changed it to `NotAllowed` because other tests are named that way.

Reply:
ok i'm fine with either way